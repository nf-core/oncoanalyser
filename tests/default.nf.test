nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    profile "test_stub"
    options "-stub"

    test("-profile test_stub") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                ).match() }
            )
        }
    }
}
