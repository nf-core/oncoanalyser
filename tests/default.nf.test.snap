{
    "-profile test_stub": {
        "content": [
            34,
            [
                "pipeline_info",
                "pipeline_info/software_versions.yml",
                "subject_a",
                "subject_a/alignments",
                "subject_a/alignments/dna",
                "subject_a/alignments/dna/subject_a.normal.duplicate_freq.tsv",
                "subject_a/alignments/dna/subject_a.normal.jitter_params.tsv",
                "subject_a/alignments/dna/subject_a.normal.ms_table.tsv.gz",
                "subject_a/alignments/dna/subject_a.normal.redux.bam",
                "subject_a/alignments/dna/subject_a.normal.redux.bam.bai",
                "subject_a/alignments/dna/subject_a.tumor.duplicate_freq.tsv",
                "subject_a/alignments/dna/subject_a.tumor.jitter_params.tsv",
                "subject_a/alignments/dna/subject_a.tumor.ms_table.tsv.gz",
                "subject_a/alignments/dna/subject_a.tumor.redux.bam",
                "subject_a/alignments/dna/subject_a.tumor.redux.bam.bai",
                "subject_a/amber",
                "subject_a/amber/placeholder",
                "subject_a/bamtools",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools/subject_a.normal.bam_metric.coverage.tsv",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools/subject_a.normal.bam_metric.flag_counts.tsv",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools/subject_a.normal.bam_metric.frag_length.tsv",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools/subject_a.normal.bam_metric.partition_stats.tsv",
                "subject_a/bamtools/subject_a_subject_a.normal_bamtools/subject_a.normal.bam_metric.summary.tsv",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools/subject_a.tumor.bam_metric.coverage.tsv",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools/subject_a.tumor.bam_metric.flag_counts.tsv",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools/subject_a.tumor.bam_metric.frag_length.tsv",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools/subject_a.tumor.bam_metric.partition_stats.tsv",
                "subject_a/bamtools/subject_a_subject_a.tumor_bamtools/subject_a.tumor.bam_metric.summary.tsv",
                "subject_a/chord",
                "subject_a/chord/subject_a.tumor.chord.mutation_contexts.tsv",
                "subject_a/chord/subject_a.tumor.chord.prediction.tsv",
                "subject_a/cider",
                "subject_a/cider/subject_a.tumor.cider.bam",
                "subject_a/cider/subject_a.tumor.cider.blastn_match.tsv.gz",
                "subject_a/cider/subject_a.tumor.cider.layout.gz",
                "subject_a/cider/subject_a.tumor.cider.locus_stats.tsv",
                "subject_a/cider/subject_a.tumor.cider.vdj.tsv.gz",
                "subject_a/cider/subject_a.tumor_rna.cider.bam",
                "subject_a/cider/subject_a.tumor_rna.cider.blastn_match.tsv.gz",
                "subject_a/cider/subject_a.tumor_rna.cider.layout.gz",
                "subject_a/cider/subject_a.tumor_rna.cider.locus_stats.tsv",
                "subject_a/cider/subject_a.tumor_rna.cider.vdj.tsv.gz",
                "subject_a/cobalt",
                "subject_a/cobalt/placeholder",
                "subject_a/cuppa",
                "subject_a/cuppa/subject_a.tumor.cuppa.pred_summ.tsv",
                "subject_a/cuppa/subject_a.tumor.cuppa.vis.png",
                "subject_a/cuppa/subject_a.tumor.cuppa.vis_data.tsv",
                "subject_a/cuppa/subject_a.tumor.cuppa_data.tsv.gz",
                "subject_a/esvee",
                "subject_a/esvee/assemble",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.alignment.tsv",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.assembly.tsv",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.breakend.tsv",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.phased_assembly.tsv",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.raw.vcf.gz",
                "subject_a/esvee/assemble/subject_a.tumor.esvee.raw.vcf.gz.tbi",
                "subject_a/esvee/caller",
                "subject_a/esvee/caller/subject_a.tumor.esvee.germline.vcf.gz",
                "subject_a/esvee/caller/subject_a.tumor.esvee.germline.vcf.gz.tbi",
                "subject_a/esvee/caller/subject_a.tumor.esvee.somatic.vcf.gz",
                "subject_a/esvee/caller/subject_a.tumor.esvee.somatic.vcf.gz.tbi",
                "subject_a/esvee/caller/subject_a.tumor.esvee.unfiltered.vcf.gz",
                "subject_a/esvee/caller/subject_a.tumor.esvee.unfiltered.vcf.gz.tbi",
                "subject_a/esvee/depth_annotation",
                "subject_a/esvee/depth_annotation/subject_a.tumor.esvee.ref_depth.vcf.gz",
                "subject_a/esvee/depth_annotation/subject_a.tumor.esvee.ref_depth.vcf.gz.tbi",
                "subject_a/esvee/prep",
                "subject_a/esvee/prep/subject_a.normal.esvee.prep.bam",
                "subject_a/esvee/prep/subject_a.normal.esvee.prep.bam.bai",
                "subject_a/esvee/prep/subject_a.tumor.esvee.prep.bam",
                "subject_a/esvee/prep/subject_a.tumor.esvee.prep.bam.bai",
                "subject_a/esvee/prep/subject_a.tumor.esvee.prep.fragment_length.tsv",
                "subject_a/esvee/prep/subject_a.tumor.esvee.prep.junction.tsv",
                "subject_a/isofox",
                "subject_a/isofox/placeholder",
                "subject_a/lilac",
                "subject_a/lilac/placeholder",
                "subject_a/linx",
                "subject_a/linx/germline_annotations",
                "subject_a/linx/germline_annotations/placeholder",
                "subject_a/linx/somatic_annotations",
                "subject_a/linx/somatic_annotations/placeholder",
                "subject_a/linx/somatic_plots",
                "subject_a/linx/somatic_plots/all",
                "subject_a/linx/somatic_plots/all/placeholder",
                "subject_a/linx/somatic_plots/reportable",
                "subject_a/linx/somatic_plots/reportable/placeholder",
                "subject_a/linx/subject_a.tumor_linx.html",
                "subject_a/orange",
                "subject_a/orange/subject_a.tumor.orange.json",
                "subject_a/orange/subject_a.tumor.orange.pdf",
                "subject_a/pave",
                "subject_a/pave/subject_a.tumor.sage.pave_germline.vcf.gz",
                "subject_a/pave/subject_a.tumor.sage.pave_germline.vcf.gz.tbi",
                "subject_a/pave/subject_a.tumor.sage.pave_somatic.vcf.gz",
                "subject_a/pave/subject_a.tumor.sage.pave_somatic.vcf.gz.tbi",
                "subject_a/peach",
                "subject_a/peach/subject_a.normal.peach.events.tsv",
                "subject_a/peach/subject_a.normal.peach.gene.events.tsv",
                "subject_a/peach/subject_a.normal.peach.haplotypes.all.tsv",
                "subject_a/peach/subject_a.normal.peach.haplotypes.best.tsv",
                "subject_a/peach/subject_a.normal.peach.qc.tsv",
                "subject_a/purple",
                "subject_a/purple/subject_a.tumor.purple.cnv.gene.tsv",
                "subject_a/purple/subject_a.tumor.purple.cnv.somatic.tsv",
                "subject_a/purple/subject_a.tumor.purple.driver.catalog.germline.tsv",
                "subject_a/purple/subject_a.tumor.purple.driver.catalog.somatic.tsv",
                "subject_a/purple/subject_a.tumor.purple.germline.vcf.gz",
                "subject_a/purple/subject_a.tumor.purple.purity.tsv",
                "subject_a/purple/subject_a.tumor.purple.qc",
                "subject_a/purple/subject_a.tumor.purple.somatic.vcf.gz",
                "subject_a/purple/subject_a.tumor.purple.sv.germline.vcf.gz",
                "subject_a/purple/subject_a.tumor.purple.sv.vcf.gz",
                "subject_a/sage",
                "subject_a/sage/append",
                "subject_a/sage/append/subject_a.normal.sage.append.vcf.gz",
                "subject_a/sage/append/subject_a.tumor.sage.append.vcf.gz",
                "subject_a/sage/germline",
                "subject_a/sage/germline/subject_a.normal.gene.coverage.tsv",
                "subject_a/sage/germline/subject_a.normal.sage.bqr.png",
                "subject_a/sage/germline/subject_a.normal.sage.bqr.tsv",
                "subject_a/sage/germline/subject_a.tumor.sage.bqr.png",
                "subject_a/sage/germline/subject_a.tumor.sage.bqr.tsv",
                "subject_a/sage/germline/subject_a.tumor.sage.germline.vcf.gz",
                "subject_a/sage/germline/subject_a.tumor.sage.germline.vcf.gz.tbi",
                "subject_a/sage/somatic",
                "subject_a/sage/somatic/subject_a.normal.sage.bqr.png",
                "subject_a/sage/somatic/subject_a.normal.sage.bqr.tsv",
                "subject_a/sage/somatic/subject_a.tumor.gene.coverage.tsv",
                "subject_a/sage/somatic/subject_a.tumor.sage.bqr.png",
                "subject_a/sage/somatic/subject_a.tumor.sage.bqr.tsv",
                "subject_a/sage/somatic/subject_a.tumor.sage.somatic.vcf.gz",
                "subject_a/sage/somatic/subject_a.tumor.sage.somatic.vcf.gz.tbi",
                "subject_a/sigs",
                "subject_a/sigs/placeholder",
                "subject_a/teal",
                "subject_a/teal/subject_a.normal.teal.telbam.bam",
                "subject_a/teal/subject_a.normal.teal.telbam.bam.bai",
                "subject_a/teal/subject_a.normal.teal.{tellength.tsv}",
                "subject_a/teal/subject_a.tumor.teal.breakend.tsv.gz",
                "subject_a/teal/subject_a.tumor.teal.telbam.bam",
                "subject_a/teal/subject_a.tumor.teal.telbam.bam.bai",
                "subject_a/teal/subject_a.tumor.teal.tellength.tsv",
                "subject_a/virusbreakend",
                "subject_a/virusbreakend/subject_a.tumor.summary.tsv",
                "subject_a/virusbreakend/subject_a.tumor.virusbreakend.vcf",
                "subject_a/virusinterpreter",
                "subject_a/virusinterpreter/subject_a.tumor.virus.annotated.tsv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-06-11T12:53:28.525304"
    }
}